contrib/platform/src/com/sun/jna/platform/win32/Crypt32.java

```diff
@@ -501,4 +501,170 @@ boolean CertVerifyCertificateChainPolicy(LPSTR pszPolicyOID, CERT_CHAIN_CONTEXT
      *      "https://msdn.microsoft.com/en-us/library/windows/desktop/aa387314(v=vs.85).aspx">MSDN</a>
      */
     HCERTSTORE PFXImportCertStore(DATA_BLOB pPFX, WTypes.LPWSTR szPassword, int dwFlags);
+
+    /**
+     * The CertEnumCertificatesInStore function retrieves the first or next
+     * certificate in a certificate store. Used in a loop, this function can
+     * retrieve in sequence all certificates in a certificate store.
+     *
+     * @param hCertStore       A handle of a certificate store.
+     * @param pPrefCertContext A pointer to the {@link CERT_CONTEXT} of the
+     *                         previous certificate context found.
+     * <p>
+     * This parameter must be NULL to begin the enumeration and get the first
+     * certificate in the store. Successive certificates are enumerated by
+     * setting {@code pPrevCertContext} to the pointer returned by a previous
+     * call to the function. This function frees the {@link CERT_CONTEXT}
+     * referenced by non-NULL values of this parameter.</p>
+     *
+     * <p>
+     * For logical stores, including collection stores, a duplicate of the
+     * pCertContext returned by this function cannot be used to begin a new
+     * subsequence of enumerations because the duplicated certificate loses the
+     * initial enumeration state. The enumeration skips any certificate
+     * previously deleted by CertDeleteCertificateFromStore.</p>
+     *
+     * @return If the function succeeds, the function returns a pointer to the
+     *         next {@link CERT_CONTEXT} in the store. If no more certificates
+     *         exist in the store, the function returns {@code NULL}.
+     *
+     * <p>
+     * For extended error information, call GetLastError. Some possible error
+     * codes follow.</p>
+     *
+     * <table>
+     * <tr><th>Value</th><th>Description</th></tr>
+     * <tr><td>E_INVALIDARG</td><td>The handle in the {@code hCertStore}
+     * parameter is not the same as that in the certificate context pointed to
+     * by {@code pPrevCertContext}.</td></tr>
+     * <tr><td>CRYPT_E_NOT_FOUND</td><td>No certificates were found. This
+     * happens if the store is empty or if the function reached the end of the
+     * store's list.</td></tr>
+     * <tr><td>ERROR_NO_MORE_FILES</td><td>Applies to external stores. No
+     * certificates were found. This happens if the store is empty or if the
+     * function reached the end of the store's list. </td></tr>
+     * </table>
+     *
+     * @see <a href=
+     *      "https://docs.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certenumcertificatesinstore">MSDN</a>
+     */
+    CERT_CONTEXT.ByReference CertEnumCertificatesInStore(HCERTSTORE hCertStore, Pointer pPrefCertContext);
```

Comment:
Typo: `Prev` (also in the `@param`). Is there a reason we don't use `CERT_CONTEXT.ByReference` for this arg?

Reply:
Typo will be fixed - I'm surprised I obviously typed the parameter and did not copy ...

For the why: Your suggestion was my first implementation, but I observed segfaults when bound like that. I reread the documentation and was able to explain the segfault:

> Successive certificates are enumerated by setting pPrevCertContext to the pointer returned by a previous call to the function. **This function frees the CERT_CONTEXT referenced by non-NULL values of this parameter**.

The problem is, that JNA reread the structure contents after the return from native and so we effectively get a "read-after-free" situation.